xapian-core/include/xapian/mset.h

```diff
@@ -74,6 +77,30 @@ class XAPIAN_VISIBILITY_DEFAULT MSet {
     /// Destructor.
     ~MSet();
 
+    /** Assigns new weights to items
+     * This functon is used in letor to update the weights assigned by the ranker
+     * It updates weights of all the elements in items as well as updating the max_attained
+     */
+    template <typename Iterator>
+    void replace_weights(Iterator first, Iterator last) {
+	if (std::distance(first, last) != size())
+	{
+	    throw std::invalid_argument("Number of weights assigned don't match the number of items.Aborting");
```

Comment:
Should be `Xapian::InvalidArgumentError`.  Also "Aborting" shouldn't be in the message (the thrower can't tell how the exception will be handled).  And the verb is wrong - "don't" should be "doesn't" since the subject is "number of weights" which is singular.